lib/src/transaction.rs

```diff
@@ -153,6 +164,9 @@ struct NewRepoData {
     index: Box<dyn ReadonlyIndex>,
```

Comment:
I'd like to add a comment to `NewRepoData` as well, but I'm not really sure why it needs to exist. It seems like it could be replaced with `ReadOnlyRepo`, which exposes the operation, view, and index.

Thoughts?

Reply:
Good question! It does seem like it could be. Try it and send a PR?

Reply:
Looks like it can be cleaned up a bit. PTAL at https://github.com/martinvonz/jj/pull/3174.